Fix calculation of page_len in sff_8436_read_write function

From: yurypm <yurypm@arista.com>


---
 drivers/misc/eeprom/sff_8436_eeprom.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/eeprom/sff_8436_eeprom.c b/drivers/misc/eeprom/sff_8436_eeprom.c
index 4688597eb..10074f03a 100644
--- a/drivers/misc/eeprom/sff_8436_eeprom.c
+++ b/drivers/misc/eeprom/sff_8436_eeprom.c
@@ -689,7 +689,7 @@ static ssize_t sff_8436_read_write(struct sff_8436_data *sff_8436,
 			if (off + pending_len < page_start_offset + SFF_8436_PAGE_SIZE) {
 				page_len = pending_len;
 			} else {
-				page_len = SFF_8436_PAGE_SIZE - off;
+				page_len = SFF_8436_PAGE_SIZE - (off - page_start_offset);
 			}
 		} else {
 			page_offset = page_start_offset;
